src/core/job.c

```diff
@@ -848,7 +848,9 @@ int job_finish_and_invalidate(Job *j, JobResult result, bool recursive) {
 
         log_unit_debug(u, "Job %s/%s finished, result=%s", u->id, job_type_to_string(t), job_result_to_string(result));
 
-        job_emit_status_message(u, t, result);
+        /* If this job did nothing to respective unit we don't log the status message */
+        if (_likely_(!already))
+                job_emit_status_message(u, t, result);
```

Comment:
I don't really like the _likely_ usage here. We don#t really know that it is more likely that the status message is printed here... Unless something is a known inner loop, or we really, really know that some if check will always fail or always succeed we shouldn't use _likely_ or _unlikely.